lib/Serialization/ModuleFile.cpp

```diff
@@ -339,8 +346,18 @@ class ModuleFile::DeclTableInfo {
   }
 
   static internal_key_type ReadKey(const uint8_t *data, unsigned length) {
-    // TODO: Handle special names
-    return StringRef(reinterpret_cast<const char *>(data), length);
+    uint8_t kind = endian::readNext<uint8_t, little, unaligned>(data);
+    switch (kind) {
+    case (uint8_t)DeclBaseName::Kind::Normal: {
```

Comment:
I guess you mean Serialization/ModuleFormat.h

I also use the casted `DeclBaseName::Kind` for serialisation in the ClangImporter. Where should I put the mirror enum here? There doesn't seem to be a central place like in Serialization. Or should I import `ModuleFormat.h`? I can't find a dependency from ClangImporter to Serialization yet, so I'm not quite sure if that is a good idea.